### ISSUE TITLE
Error: local variable 'runtime_image_tag' referenced before assignment when runnning sam local invoke with --invoke-image and AWS layers

### ISSUE THREAD

**Issue by francisoud** [aws/aws-sam-cli#4949]
### Description:
`Error: local variable 'runtime_image_tag' referenced before assignment`
when using AWS layers and
sam local invoke with --invoke-image 
same error with start-api

### Steps to reproduce:

`docker login docker-virtual.artifactory.corp`

```
sam init --name sam-hello-layers --runtime python3.9 --app-template hello-world
Would you like to enable X-Ray tracing on the function(s) in your application?  [y/N]: N

Would you like to enable monitoring using CloudWatch Application Insights?
For more info, please view https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/cloudwatch-application-insights.html [y/N]: N
```

Edit template.yaml and a layer:

```
Globals:
  Function:
    Timeout: 3
    MemorySize: 128
    Layers:  
      - !Sub arn:aws:lambda:${AWS::Region}:017000801446:layer:AWSLambdaPowertoolsPythonV2:24
```

`sam build --region eu-west-1`

Login to aws: aws configure...

### Observed result:
`sam local invoke HelloWorldFunction --event events/event.json --invoke-image docker-virtual.artifactory.corp/sam/emulation-python3.9 --region eu-west-1 --debug`
Output:
```
2023-03-30 15:52:52,439 | Config file location: /home/benjamin/test-sam-python/sam-hello-layers/samconfig.toml
2023-03-30 15:52:52,439 | Loading configuration values from [default.['local', 'invoke'].parameters] (env.command_name.section) in config file at '/home/benjamin/test-sam-python/sam-hello-layers/samconfig.toml'...
2023-03-30 15:52:52,444 | Configuration values successfully loaded.
2023-03-30 15:52:52,444 | Configuration values are: {'stack_name': 'sam-hello-layers'}
2023-03-30 15:52:52,448 | Using SAM Template at /home/benjamin/test-sam-python/sam-hello-layers/.aws-sam/build/template.yaml
2023-03-30 15:52:52,507 | Using config file: samconfig.toml, config environment: default
2023-03-30 15:52:52,507 | Expand command line arguments to:
2023-03-30 15:52:52,508 | --template_file=/home/benjamin/test-sam-python/sam-hello-layers/.aws-sam/build/template.yaml --event=events/event.json --invoke_image=('docker-virtual.artifactory.corp/sam/emulation-python3.9',) --function_logical_id=HelloWorldFunction --no_event --layer_cache_basedir=/home/benjamin/.aws-sam/layers-pkg --container_host=localhost --container_host_interface=127.0.0.1 
2023-03-30 15:52:52,508 | local invoke command is called
2023-03-30 15:52:52,512 | No Parameters detected in the template
2023-03-30 15:52:52,524 | Sam customer defined id is more priority than other IDs. Customer defined id for resource HelloWorldFunction is HelloWorldFunction
2023-03-30 15:52:52,525 | There is no customer defined id or cdk path defined for resource ServerlessRestApi, so we will use the resource logical id as the resource id
2023-03-30 15:52:52,525 | 0 stacks found in the template
2023-03-30 15:52:52,526 | No Parameters detected in the template
2023-03-30 15:52:52,536 | Sam customer defined id is more priority than other IDs. Customer defined id for resource HelloWorldFunction is HelloWorldFunction
2023-03-30 15:52:52,536 | There is no customer defined id or cdk path defined for resource ServerlessRestApi, so we will use the resource logical id as the resource id
2023-03-30 15:52:52,537 | 2 resources found in the stack 
2023-03-30 15:52:52,537 | Found Serverless function with name='HelloWorldFunction' and CodeUri='HelloWorldFunction'
2023-03-30 15:52:52,537 | --base-dir is not presented, adjusting uri HelloWorldFunction relative to /home/benjamin/test-sam-python/sam-hello-layers/.aws-sam/build/template.yaml
2023-03-30 15:52:52,545 | Found one Lambda function with name 'HelloWorldFunction'
2023-03-30 15:52:52,545 | Invoking app.lambda_handler (python3.9)
2023-03-30 15:52:52,545 | No environment variables found for function 'HelloWorldFunction'
2023-03-30 15:52:52,546 | Loading AWS credentials from session with profile 'None'
2023-03-30 15:52:53,251 | Resolving code path. Cwd=/home/benjamin/test-sam-python/sam-hello-layers/.aws-sam/build, CodeUri=/home/benjamin/test-sam-python/sam-hello-layers/.aws-sam/build/HelloWorldFunction
2023-03-30 15:52:53,251 | Resolved absolute path to code is /home/benjamin/test-sam-python/sam-hello-layers/.aws-sam/build/HelloWorldFunction
2023-03-30 15:52:53,251 | Code /home/benjamin/test-sam-python/sam-hello-layers/.aws-sam/build/HelloWorldFunction is not a zip/jar file
2023-03-30 15:52:53,251 | arn:aws:lambda:eu-west-1:017000801446:layer:AWSLambdaPowertoolsPythonV2:24 is already cached. Skipping download
2023-03-30 15:52:53,251 | Cleaning all decompressed code dirs
2023-03-30 15:52:53,252 | Telemetry endpoint configured to be https://aws-serverless-tools-telemetry.us-west-2.amazonaws.com/metrics

Error: local variable 'runtime_image_tag' referenced before assignment
Traceback:
  File "click/core.py", line 1055, in main
  File "click/core.py", line 1657, in invoke
  File "click/core.py", line 1657, in invoke
  File "click/core.py", line 1404, in invoke
  File "click/core.py", line 760, in invoke
  File "click/decorators.py", line 84, in new_func
  File "click/core.py", line 760, in invoke
  File "samcli/lib/telemetry/metric.py", line 183, in wrapped
  File "samcli/lib/telemetry/metric.py", line 148, in wrapped
  File "samcli/lib/utils/version_checker.py", line 42, in wrapped
  File "samcli/cli/main.py", line 92, in wrapper
  File "samcli/commands/local/invoke/cli.py", line 118, in cli
  File "samcli/commands/local/invoke/cli.py", line 203, in do_cli
  File "samcli/commands/local/lib/local_lambda.py", line 151, in invoke
  File "samcli/lib/telemetry/metric.py", line 323, in wrapped_func
  File "samcli/local/lambdafn/runtime.py", line 189, in invoke
  File "samcli/local/lambdafn/runtime.py", line 100, in create
  File "samcli/local/docker/lambda_container.py", line 95, in __init__
  File "samcli/local/docker/lambda_container.py", line 237, in _get_image
  File "samcli/local/docker/lambda_image.py", line 181, in build
```

Same error with sam build --use-container --build-image docker-virtual.artifactory.corp/sam/build-python3.9 --region eu-west-1

### Expected result:
invoke HelloFunction sucessful

### Additional environment details (Ex: Windows, Mac, Amazon Linux etc)
in WSL with Ubuntu 22
and 
python --version
Python 3.9.16
sam --version
SAM CLI, version 1.78.0

```
sam --info
{
  "version": "1.78.0",
  "system": {
    "python": "3.7.10",
    "os": "Linux-5.15.90.1-microsoft-standard-WSL2-x86_64-with-debian-bookworm-sid"
  },
  "additional_dependencies": {
    "docker_engine": "20.10.21",
    "aws_cdk": "Not available",
    "terraform": "Not available"
  }
}
```

**Comment by mildaniel** [comment]
Hi @francisoud, thanks for opening this issue. I confirmed the bug and raised a PR with a fix.

**Comment by qingchm** [comment]
Hey there, this has been addressed in our most recent release 1.79.0 (https://github.com/aws/aws-sam-cli/releases/tag/v1.79.0). Feel free to check it our. Closing the issue here. Thanks for cutting the issue!

**Comment by francisoud** [comment]
I confirm this is fixed in v1.79
Thanks!
